src/domain/individualHarbor/forms/berthForm/BerthForm.tsx

```diff
@@ -0,0 +1,192 @@
+import React from 'react';
+import { Formik } from 'formik';
+import * as Yup from 'yup';
+import { useTranslation } from 'react-i18next';
+import { TFunction } from 'i18next';
+import { Button, TextInput } from 'hds-react';
+import { ObjectSchema } from 'yup';
+
+import { Berth, FormProps } from './types';
+import { BerthMooringType } from '../../../../@types/__generated__/globalTypes';
+import Grid from '../../../../common/grid/Grid';
+import Select from '../../../../common/select/Select';
+import Checkbox from '../../../../common/checkbox/Checkbox';
+import styles from './berthForm.module.scss';
+import { Pier } from '../../utils/utils';
+import Text from '../../../../common/text/Text';
+
+interface BerthFormProps extends FormProps<Berth> {
+  onSubmitText?: string;
+  pierOptions: Pier[];
+}
+
+const getBerthValidationSchema = (
+  t: TFunction,
+  pierOptions: Pier[]
+): ObjectSchema => {
+  return Yup.object().shape({
+    pierId: Yup.string()
+      .oneOf(pierOptions.map(pier => pier.id))
+      .required(t('forms.common.errors.required')),
+    number: Yup.number()
```

Comment:
Should we specify that all of these numbers must be greater than zero?

Reply:
Added `positive` and `integer` validators to applicable fields.